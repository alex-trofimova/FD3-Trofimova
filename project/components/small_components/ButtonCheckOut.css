.Btn_check_out {
    background-color: rgb(255, 165, 0);
    padding: 5px 20px;
    font-size: 16px;
    border: 1px solid rgb(182, 92, 8); 
    border-radius: 5px;
    min-width: 100px;
}

.Btn_check_out:focus {
    border: none;
}

.Btn_check_out:hover {
    background-color:rgb(255, 150, 0);
  }