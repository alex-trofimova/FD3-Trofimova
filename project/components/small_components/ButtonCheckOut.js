import React from 'react';
import PropTypes from 'prop-types';

import './ButtonCheckOut.css';

class ButtonCheckOut extends React.PureComponent {

static propTypes = {
    message: PropTypes.string.isRequired,
};

showMessage = ()=> {
    alert('Ваша '+this.props.message+' оформлена');
}

  render() {
    let btn_title = (this.props.message==='покупка')?'заказ':'заявку';

    return (
        <button className="Btn_check_out" onClick={this.showMessage}>
            {'Оформить '+btn_title} 
        </button>      
    )
    ;

  }
}

export default ButtonCheckOut;
