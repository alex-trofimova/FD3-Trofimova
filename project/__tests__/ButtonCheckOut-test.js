import React from 'react';
import renderer from 'react-test-renderer';

import ButtonCheckOut from '../components/small_components/ButtonCheckOut';

test('работа ButtonCheckOut', () => {

  const component = renderer.create(
    <ButtonCheckOut message='покупка'/>
  );

  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  //нажатие на кнопку
  const buttonElem = component.root.find(  el => el.props.className == 'Btn_check_out' );
  buttonElem.props.onClick();

  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

    
});
