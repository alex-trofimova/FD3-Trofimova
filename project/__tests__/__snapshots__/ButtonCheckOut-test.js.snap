// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа ButtonCheckOut 1`] = `
<button
  className="Btn_check_out"
  onClick={[Function]}
>
  Оформить заказ
</button>
`;

exports[`работа ButtonCheckOut 2`] = `
<button
  className="Btn_check_out"
  onClick={[Function]}
>
  Оформить заказ
</button>
`;
